modules/test/playwright/tests/frontend-data-set-views-web/settings.spec.ts

```diff
@@ -0,0 +1,163 @@
+ */
+
+import {expect, mergeTests} from '@playwright/test';
+
+import {featureFlagsTest} from '../../fixtures/featureFlagsTest';
+import {loginTest} from '../../fixtures/loginTest';
+import {dataSetManagerApiHelpersTest} from './fixtures/dataSetManagerApiHelpersTest';
+import {dataSetsPageTest} from './fixtures/dataSetsPageTest';
+import {fieldsPageTest} from './fixtures/fieldsPageTest';
+import {settingsPageTest} from './fixtures/settingsPageTest';
+import {visualizationModesPageTest} from './fixtures/visualizationModesPageTest';
+import {DEFAULT_LABEL} from './utils/constants';
+
+export const test = mergeTests(
+	dataSetsPageTest,
+	dataSetManagerApiHelpersTest,
+	featureFlagsTest({
+		'LPD-10735': true,
+		'LPS-164563': true,
+		'LPS-178052': true,
+	}),
+	fieldsPageTest,
+	loginTest(),
+	settingsPageTest,
+	visualizationModesPageTest
+);
+
+test.beforeEach(async ({dataSetManagerApiHelpers}) => {
+	await dataSetManagerApiHelpers.createDataSet({erc: 'SettingsDataSetERC'});
+	await dataSetManagerApiHelpers.createDataSetView({
+		r_fdsEntryFDSViewRelationship_c_fdsEntryERC: 'SettingsDataSetERC',
+	});
+});
+
+test.afterEach(async ({dataSetManagerApiHelpers}) => {
+	await dataSetManagerApiHelpers.deleteDataSet({erc: 'SettingsDataSetERC'});
+});
+
+test.describe('Data Set Settings', () => {
+	test.describe('Default Visualization Mode', () => {
+		test('If Default Visualization Mode is not configured allows user to navigate to Visualization Mode section', async ({
+			settingsPage,
+			visualizationModesPage,
+		}) => {
+			await test.step('Navigate to Settings section', async () => {
+				await settingsPage.goto({
+					dataSetLabel: DEFAULT_LABEL.DATA_SET,
+					viewLabel: DEFAULT_LABEL.VIEW,
+				});
+
+				await settingsPage.defaultVisualizationModeLabel.isVisible();
+			});
+
+			await test.step('Check Default Visualization Mode', async () => {
+				await expect(
+					settingsPage.defaultVisualizationModeLabel
+				).toContainText('Not Configured');
+			});
+
+			await test.step('Navigate to Visualization Mode section if  "Not Configured"', async () => {
+				await settingsPage.defaultVisualizationModeLabel.click();
+
+				await settingsPage.configureNewLayoutButton.isVisible();
+
+				await settingsPage.configureNewLayoutButton.click();
+
+				await visualizationModesPage.page
+					.getByText('No fields added yet.')
+					.isVisible();
+			});
+		});
+
+		test('When there is only one visualization mode defined, that will be the default one.', async ({
+			dataSetManagerApiHelpers,
+			settingsPage,
+		}) => {
+			await test.step('Assign a field to a Card title section', async () => {
+				await dataSetManagerApiHelpers.createDataSetViewCardsSection(
+					{}
+				);
+			});
+
+			await test.step('Navigate to Settings section', async () => {
+				await settingsPage.goto({
+					dataSetLabel: DEFAULT_LABEL.DATA_SET,
+					viewLabel: DEFAULT_LABEL.VIEW,
+				});
+
+				await settingsPage.defaultVisualizationModeLabel.isVisible();
```

Comment:
Caution with `isVisible`, it may result in flakiness.

Reply:
@markocikos Will change by `expec(..).toBeVisible()` as recommended by [Playwright docs](https://playwright.dev/docs/api/class-locator#locator-is-visible)

Reply:
I'd recommend `expect(..).toBeInViewport()`, although I'm not sure if there is an even better solution. I found `expect(..).toBeVisible()` to sometimes gives false positives. It [checks these things](https://playwright.dev/docs/actionability#visible), but few times for me, this was not enough.

Reply:
Ok, I'll try with `toBeInViewport()`